workflow/config/config.go

```diff
@@ -108,7 +108,7 @@ type PostgreSQLConfig struct {
 	TableName      string                  `json:"tableName"`
 	UsernameSecret apiv1.SecretKeySelector `json:"userNameSecret"`
 	PasswordSecret apiv1.SecretKeySelector `json:"passwordSecret"`
-	SSL            bool                    `json:"ssl,omitempty"`
+	SSLMode        string                  `json:"sslMode,omitempty"`
```

Comment:
We are technically changing an API here, but since it has been a short time and a niche feature I think it might be fine.

This is in @sarabala1979's domain so I will let him make the final review.

Reply:
Ah yeah didn't think of that. Happy to change it if folk are already using it. Is there some documentation/guidance on managing API changes for the Argo project somewhere I can refer to in order to get it right next time?

Reply:
Can we please have two flags? You can mark `SSL` as deprecated.

Reply:
Yep will do. Do you guys manage deprecation through a comment in the code or is there a way to surface it back to the OpenAPI spec?

Reply:
I think you make them with the commend `//DEPRECATED`, but how about using `SSL` to turn it on and `SSLMode` to choose the mode?

Reply:
I'm not fussed either way. Which are you most comfortable with?

Reply:
lets have both fields

Reply:
On it. Will send a patch in the next few hours.

Reply:
Edit: I patched it in a few months...